src/lib/checkbox/checkbox.ts

```diff
@@ -164,6 +164,13 @@ export class MdCheckbox implements ControlValueAccessor, AfterViewInit, OnDestro
 
   @ViewChild(MdRipple) _ripple: MdRipple;
 
+  @ViewChild('labelWrapper') _labelWrapper: ElementRef;
+
+  get hasLabel(): boolean {
+    const labelWrapperTextContent = this._labelWrapper.nativeElement.textContent;
```

Comment:
This will be a problem for Angular Universal. Currently, `nativeElement` is a parse5 node, which doesn't have a `textContent` property. So for now you can say
```ts
const labelText = this._labelWrapper.nativeElement.textContent || '';
return !!labelText.trim().length;
```